miniapps/nurbs/nurbs_naca_cmesh.cpp

```diff
@@ -0,0 +1,792 @@
+   output << "elements"<<endl;
+   output << "5"<<endl;
+   output << "1 3 0 1 5 4" << endl;   // Lower wake
+   output << "1 3 1 2 6 5" << endl;   // Lower tail
+   output << "1 3 2 3 7 6" << endl;   // Tip
+   output << "1 3 3 1 8 7" << endl;   // Upper tail
+   output << "1 3 1 0 9 8" << endl;   // Upper wake
+   output << endl;
+
+   // Boundaries
+   output << "boundary" <<endl;
+   output << "10" <<endl;
+   output << "1 1 5 4" << endl;   // Bottom
+   output << "1 1 6 5" << endl;   // Bottom
+   output << "2 1 7 6" << endl;   // Inflow
+   output << "3 1 8 7" << endl;   // Top
+   output << "3 1 9 8" << endl;   // Top
+   output << "4 1 4 0" << endl;   // Outflow
+   output << "4 1 0 9" << endl;   // Outflow
+   output << "5 1 1 2" << endl;   // Foil section
+   output << "5 1 2 3" << endl;   // Foil section
+   output << "5 1 3 1" << endl;   // Foil section
+   output << endl;
+
+   // Edges
+   output <<"edges"<<endl;
+   output <<"15"<<endl;
+   output << "0 0 1"<<endl;
+   output << "1 1 2"<<endl;
+   output << "2 2 3"<<endl;
+   output << "3 3 1"<<endl;
+
+   output << "0 4 5"<<endl;
+   output << "1 5 6"<<endl;
+   output << "2 6 7"<<endl;
+   output << "3 7 8"<<endl;
+   output << "0 9 8"<<endl;
+
+   output << "4 0 4"<<endl;
+   output << "4 1 5"<<endl;
+   output << "4 2 6"<<endl;
+   output << "4 3 7"<<endl;
+   output << "4 1 8"<<endl;
+   output << "4 0 9"<<endl;
+   output << endl;
+
+   // Vertices
+   output << "vertices" << endl;
+   output << 10 << endl;
+   output << endl;
+
+   // Patches
+   output<<"patches"<<endl;
+   output<<endl;
+
+   output << "# Patch 0 " << endl;
+   patch0.Print(output); output<<endl;
+   output << "# Patch 1 " << endl;
+   patch1.Print(output); output<<endl;
+   output << "# Patch 2 " << endl;
+   patch2.Print(output); output<<endl;
+   output << "# Patch 3 " << endl;
+   patch3.Print(output); output<<endl;
+   output << "# Patch 4 " << endl;
+   patch4.Print(output); output<<endl;
+
+   // Close
+   output.close();
+   delete kv0;
+   delete kv1;
+   delete kv2;
+   delete kv3;
+   delete kv4;
+   delete kvr;
+   delete kv_o1;
+   delete kv_o2;
+   delete xyf[0];
+   delete xyf[1];
+
+   cout << endl << "Boundary identifiers:" << endl;
+   cout << "   1   Bottom" << endl;
+   cout << "   2   Inflow" << endl;
+   cout << "   3   Top" << endl;
+   cout << "   4   Outflow" << endl;
+   cout << "   5   Foil section" << endl;
+   cout << "=========================================================="<< endl;
+   cout << "  "<< mdim <<"D mesh generated: " <<mesh_file.c_str()<< endl ;
+   cout << "=========================================================="<< endl;
+
+   // Print mesh info to screen
+   cout << "=========================================================="<< endl;
+   cout << " Attempting to read mesh: " <<mesh_file.c_str()<< endl ;
+   cout << "=========================================================="<< endl;
+   Mesh *mesh = new Mesh(mesh_file.c_str(), 1, 1);
+   mesh->PrintInfo();
```

Comment:
`mesh->Print(os);` for an `ofstream os` will write a file identical to the one in `Naca_cmesh/mesh_000000`, which can be visualized with glvis. The format of that file is different from `naca-cmesh.mesh`, but I'm not sure if there is a real difference in the loaded meshes.